lib/HLSL/DxilValidation.cpp

```diff
@@ -723,7 +730,9 @@ struct ValidationContext {
       }
       DiagPrinter << "#" << idx;
```

Comment:
I just noticed that this block number `idx` is never incremented in the loop, making it always print `#0` for unnamed blocks.

Reply:
Good catch!